### PR TITLE
update doc on wal-recovery-mode and use-direct-io-for-flush-and-compaction

### PR DESCRIPTION
**Pull request by ekexium**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Complement the descriptions of wal-recovery-mode and use-direct-io-for-flush-and-compaction.
Previous descriptions are vague.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6860
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -718,6 +718,10 @@ Configuration items related to RocksDB
 
 + WAL recovery mode
 + Optional values: `0` (`TolerateCorruptedTailRecords`), `1` (`AbsoluteConsistency`), `2` (`PointInTimeRecovery`), `3` (`SkipAnyCorruptedRecords`)
++ `0` (`TolerateCorruptedTailRecords`): tolerate incomplete record in trailing data on all logs
++ `1` (`AbsoluteConsistency`): we don't expect to find any corruption in the WAL
++ `2` (`PointInTimeRecovery`): recover to point-in-time consistency
++ `3` (`SkipAnyCorruptedRecords`): recovery after a disaster
```
Comment (TomShawn) [inline tikv-configuration-file.md:724]: @ekexium Please help sync the changes made in https://github.com/pingcap/docs-cn/pull/6860/files#diff-5e0b63d087f45ab3370aeb3f520d8d1f637a6f031fb6b9f294b4addc227e24c5R760-R765 to here. Thanks~

**Review by Connor1996 — APPROVED** [review]
LGTM

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -718,6 +718,10 @@ Configuration items related to RocksDB
 
 + WAL recovery mode
 + Optional values: `0` (`TolerateCorruptedTailRecords`), `1` (`AbsoluteConsistency`), `2` (`PointInTimeRecovery`), `3` (`SkipAnyCorruptedRecords`)
```
Comment (TomShawn) [inline tikv-configuration-file.md:720]:
```suggestion
+ WAL recovery mode
+ Value options: `0`, `1`, `2`, `3`
```

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -718,6 +718,10 @@ Configuration items related to RocksDB
...
++ `0` (`TolerateCorruptedTailRecords`): tolerate and discard incomplete record in trailing data on all logs
```
Comment (TomShawn) [inline tikv-configuration-file.md:721]:
```suggestion
+ `0` (`TolerateCorruptedTailRecords`): Tolerates and discards the records that have incomplete trailling data on all logs.
```

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -718,6 +718,10 @@ Configuration items related to RocksDB
...
++ `1` (`AbsoluteConsistency`): abandon the recovery when corrupted logs are found
```
Comment (TomShawn) [inline tikv-configuration-file.md:722]:
```suggestion
+ `1` (`AbsoluteConsistency`): abandons recovery when corrupted logs are found.
```

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -718,6 +718,10 @@ Configuration items related to RocksDB
...
++ `2` (`PointInTimeRecovery`): recover sequentially until the first corrupted log is encoutered
```
Comment (TomShawn) [inline tikv-configuration-file.md:723]:
```suggestion
+ `2` (`PointInTimeRecovery`): recovers log sequentially until the first corrupted log is encountered.
```

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -767,7 +771,7 @@ Configuration items related to RocksDB
 
 ### `use-direct-io-for-flush-and-compaction`
 
-+ Determines whether to use `O_DIRECT` for both reads and writes in background flush and compactions
++ Determines whether to use `O_DIRECT` for both reads and writes in background flush and compactions. The performance impact of this option: turning on O_DIRECT bypasses and prevents contamination of the OS buffer cache, but subsequent file reads require re-reading the contents to the buffer cache.
```
Comment (TomShawn) [inline tikv-configuration-file.md:774]:
```suggestion
+ Determines whether to use `O_DIRECT` for both reads and writes in the background flush and compactions. The performance impact of this option: enabling `O_DIRECT` bypasses and prevents contamination of the OS buffer cache, but the subsequent file reads require re-reading the contents to the buffer cache.
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**